src/api/Bolt.tsx

```diff
@@ -0,0 +1,243 @@
+    //     }
+    // }
+
+    /**
+     * Removes a panel from its window and removes all of its bindings.
+     * The panel and its bindings should not be used again
+     * @param panel - panel to dispose
+     */
+    // public disposePanel(panel: PanelController): void
+    // {
+    //     this.removePanel(panel);
+    //     panel._dispose();
+    // }
+
+    /**
+     * Removes all panels from the gui. All binding are also disposed of.
+     * The panels, and bindings should not be used again
+     */
+    // public disposeAllPanels(): void
+    // {
+    //     const panels = this._panels.splice(0, this._panels.length);
+
+    //     for (let i = 0; i < panels.length; ++i)
+    //     {
+    //         panels[i]._dispose();
+    //     }
+
+    //     this._panels = [];
+    // }
+
+    /**
+     * Removes all panels and windows from the gui. All binding are also disposed of.
+     * The windows, panels, and bindings should not be used again
+     */
+    // public dispose(): void
+    // {
+    //     this.disposeAllPanels();
+    //     this._windows = [];
+    //     this._defaultWindow = this.createWindow();
+    // }
+
+    /**
+     * Creates a new window in which you can add panels too.
+     * @param options - options for the window
+     */
+    public createWindow(options: WindowData): WindowData
+    {
+        store.dispatch(addWindow(options));
+
+        return options;
+    }
+
+    /**
+     * Creates a new component
+     * @param options - options for the window
+     */
+    public createComponent(type: string, id: string, panelId: string, inputData: any): any
+    {
+        switch (type)
+        {
+            case 'label': {
+                store.dispatch(addComponent({ id, component: Label, inputData }, panelId));
+                break;
+            }
+            case 'button': {
+                store.dispatch(addComponent({ id, component: ButtonInputComponent, inputData }, panelId));
+                break;
+            }
+            case 'xyr': {
+                break;
+            }
+            case 'custom': {
+                break;
+            }
+            default: {
+                break;
+            }
+        }
+
+        // store.dispatch(addWindow(options));
+
+        return inputData;
+    }
+
+    /**
+     * updates component
+     */
+    public updateComponent(componentData: ComponentPair<any>): ComponentPair<any>
+    {
+        store.dispatch(updateComponent(componentData));
+
+        return componentData;
+    }
+
+    /**
+     * removes component
+     */
+    public removeComponent(id: string, parentId: string): void
+    {
+        store.dispatch(removeComponent(id, parentId));
+    }
+
+    /**
+     * returns default window data
+     */
+    public getDefaultWindowData(): WindowData { return this._defaultWindow; }
+
+    /**
+     * Refreshes the binding for each input
+     */
+    public refresh(): void
+    {
+        // refreshes the input bindings
```

Comment:
useless function

Reply:
it is not implemented yet, and will not be in the scope of this PR, however it was causing issues with the react compiler that were preventing me from building/running the project. Commented out until implementation is completed.